library/src/test/java/com/novoda/downloadmanager/lib/CriteriaTest.java

```diff
@@ -0,0 +1,180 @@
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_AND_ANOTHER_SELECTION_COLUMN_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenTwoSelectionsWhenWeBuildWithOrThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_OR_ANOTHER_SELECTION_COLUMN_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWhenWeBuildThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .and()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_OR_ANOTHER_SELECTION_COLUMN_QUERY + AND + SELECTION_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWithInnerCriteriaWhenWeBuildThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withInnerCriteria(
+                        new Criteria.Builder()
+                                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                                .withArgument(ARGUMENT)
+                                .and()
+                                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                                .withArgument(ARGUMENT)
+                                .build())
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_QUERY + OR + "(" + ANOTHER_SELECTION_COLUMN_QUERY + AND + SELECTION_QUERY + ")")
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWithMultipleInnerCriteriaWhenBuildingThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withInnerCriteria(
```

Comment:
@Electryc I don't care about the naming but it should be easy to understand what it can be used for.